pkg/registry/admissionregistration/mutatingwebhookconfiguration/strategy_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutatingwebhookconfiguration
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/apis/admissionregistration"
+)
+
+func Test_mutatingWebhookConfigurationStrategy_PrepareForCreate(t *testing.T) {
+	type args struct {
+		ctx                context.Context
+		obj                runtime.Object
+		expectedGeneration int64
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "generation is set to 1 when the configuration is prepared for creation",
+			args: args{
+				ctx:                context.TODO(),
+				obj:                &admissionregistration.MutatingWebhookConfiguration{},
+				expectedGeneration: 1,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := mutatingWebhookConfigurationStrategy{
+				ObjectTyper:   nil,
+				NameGenerator: nil,
+			}
+			m.PrepareForCreate(tt.args.ctx, tt.args.obj)
+			v, ok := tt.args.obj.(*admissionregistration.MutatingWebhookConfiguration)
+			if !ok {
+				t.Errorf("PrepareForCreate: error casting runtime.Object to (*admissionregistration.MutatingWebhookConfiguration)")
```

Comment:
minor nit: `PrepareForCreate: ` should not be needed as go shows which test fails.
[1]


also, should be t.Fatalf() ?